### PR TITLE
feat: Probot v11

### PR DESCRIPTION
**Pull request by jetersen**

### CODE REVIEW

**Comment by jetersen** [comment]
ah tests are misleading as they are mocked :sweat:

**Comment by gr2m** [comment]
> ah tests are misleading as they are mocked 😓

yeah that is a problem I've seen in many apps. Only mock http requests, never mock the methods.

Anyway, I'd like to update the adapter based on the latest Probot version, so we can use it in the example:
https://github.com/probot/example-github-action/blob/73ba8af469c1ebb4ce7574ed2c14bdaad9b781e4/index.js

I think the example code should look something like this

```
const { runProbot, createProbot } = require("@probot/actions-adapter")
const app = require('./app')

runProbot(app, { probot: createProbot })
```

`createProbot` should preset the token as it's currently done in the example, and `runProbot` should do the checks for the environment variables.

Would you like to give that a try?

Once we update the adapter, we could also create an official `probot/action` that you can use directly in your workflows, with the path to your `app.js` Probot app function.

**Comment by jetersen** [comment]
something like this: [`4ade1d3`](https://github.com/probot/actions-adapter/pull/16/commits/4ade1d3246c72e4b000b364bd424ced4219be64b)

**Comment by jetersen** [comment]
Noted in testing that `createProbot` does not log properly to github actions 😓

**Comment by gr2m** [comment]
> something like this: [`4ade1d3`](https://github.com/probot/actions-adapter/pull/16/commits/4ade1d3246c72e4b000b364bd424ced4219be64b)

looks good 👍🏼 

> Right the default log level in `createProbot` is warn.

Maybe it would make sense to change the default logging behavior for GitHub Actions, utilizing [`@actions/core`](https://github.com/actions/toolkit/tree/master/packages/core/). Probot is using [`pino`](https://getpino.io/) for logging, I think the best approach would be to create a transport for github actions, unless one already exists? The benefit is that the logging will be better integrated with GitHub Action's built-in feature.

What do you think? Would you like to give that a go?

File: index.js [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
 
-module.exports = (...handlers) => {
-  // Setup Probot app
-  const githubToken = process.env.GITHUB_TOKEN;
-  const probot = new Probot({ githubToken });
-  probot.setup(handlers);
+function createProbot({
+  overrides = {},
+  defaults = {},
+  env = process.env,
```
Comment (elstudio) [inline index.js:12]: I think the comment at the end of this line is what the linter is complaining about.

**Comment by gr2m** [comment]
superseded by https://github.com/probot/adapter-github-actions/pull/19

**Comment by jetersen** [comment]
Thanks for taking over 😅

**Pull request closed without merging**